### PR TITLE
refactor: update dependence and remove deprecated api

### PR DESCRIPTION
**Pull request by tohsaka888**
**dependence**:
- remove `@types/mongodb`
- update `mongodb`, `node-ts-scripts`

**feat**:
- add example

**fix**
- fix type error
- remove deprecated api

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,11 +12,10 @@
     "build": "node-ts-scripts bundle src/index.ts --dts --format=commonjs --platform=node --outputDir=build"
   },
   "devDependencies": {
-    "node-ts-scripts": "^0.12.17"
+    "node-ts-scripts": "^0.27.2"
   },
   "dependencies": {
-    "@types/mongodb": "^3.5.27",
-    "mongodb": "^3.6.1"
+    "mongodb": "^4.9.0"
```
Comment (heineiuo) [inline package.json]: Move `mongodb` to `peerDependencies`( version >= 4.9 or >=4.0 ) so we don't need to upgrading this anymore.

File: example/index.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { Db } from 'mongodb'
+
+const mongoUri = process.env.mongoUri as string
+const dbName = process.env.dbName as string
+const collectionName = process.env.collectionName as string
+
+const connectDB = async (dbName: string): Promise<Db> => {
+  const db = new MongoSessionDaemon(mongoUri)
+  const session = await db.getSession()
+  return await session.db(dbName)
```
Comment (heineiuo) [inline example/index.ts]: `session.db()` returns `Db` directly so this `await` is unnecessary.

**Review by heineiuo — CHANGES_REQUESTED** [review]

File: example/index.ts (hunk 2 above)
Comment (tohsaka888) [reply example/index.ts]: [fix unnecessary `await`](https://github.com/heineiuo/mongo-client-daemon/pull/1/commits/6678374f7f660e9c1a4a351e0b81beff8ad4b369)

File: package.json (hunk 1 above)
Comment (tohsaka888) [reply package.json]: [move `mongodb` to `peerDependencies`](https://github.com/heineiuo/mongo-client-daemon/pull/1/commits/32aa0d4130ba3046ab7f444e513e58d505a2de6f)

**Review by heineiuo — APPROVED** [review]
LGTM

**Pull request merged**